src/Uno.UI/UI/Xaml/Internal/PointerCapture.cs

```diff
@@ -0,0 +1,346 @@
+			// ** This is an IMPORTANT safety catch to prevent the application to become unresponsive **
+			Clear();
+
+			return true;
+		}
+		else if (_targets.TryGetValue(element, out var target))
+		{
+			Update(target, args);
+
+			return true;
+		}
+		else if (IsImplicitOnly)
+		{
+			// If the capture is implicit, we should not filter out events for children elements.
+
+			return true;
+		}
+		else
+		{
+			// We should dispatch the event only if the control which has captured the pointer has already dispatched the event
+			// (Which actually means that the current control is a parent of the control which has captured the pointer)
+			// Remarks: This is not enough to determine parent-child relationship when we dispatch multiple events base on the same native event,
+			//			(as they will all have the same FrameId), however in that case we dispatch events layer per layer
+			//			instead of bubbling a single event before raising the next one, so we are safe.
+			//			The only limitation would be when mixing native vs. managed bubbling, but this check only prevents
+			//			the leaf of the tree to raise the event, so we cannot mix bubbling mode in that case.
+			return MostRecentDispatchedEventFrameId >= args.FrameId;
+		}
+	}
+
+	private void Update(PointerCaptureTarget target, PointerRoutedEventArgs args)
+	{
+		target.LastDispatched = args;
+		if (MostRecentDispatchedEventFrameId < args.FrameId)
+		{
+			MostRecentDispatchedEventFrameId = args.FrameId;
+		}
+	}
+
+	private void EnsureEffectiveCaptureState()
+	{
+		if (_targets.Any())
+		{
+			// We have some target, self enable us
+
+			if (_actives.TryGetValue(Pointer.UniqueId, out var capture))
+			{
+				if (capture != this)
+				{
+					throw new InvalidOperationException("There is already another active capture.");
+				}
+			}
+			else
+			{
+				// This is what makes this capture active
+				_actives.Add(Pointer.UniqueId, this);
+			}
+
+			if (_nativeCaptureElement == null)
+			{
+				_nativeCaptureElement = _targets.Single().Value.NativeCaptureElement;
+
+				CapturePointerNative();
+			}
+		}
+		else
+		{
+			// We no longer have any target, cleanup
+
+			if (_nativeCaptureElement != null)
+			{
+				ReleasePointerNative();
+
+				_nativeCaptureElement = null;
+			}
+
+			if (_actives.TryGetValue(Pointer.UniqueId, out var capture) && capture == this)
+			{
+				// This is what makes this capture inactive
+				_actives.Remove(Pointer.UniqueId);
+			}
+		}
+	}
+
+	/// <remarks>
+	/// This method contains or is called by a try/catch containing method and can
+	/// be significantly slower than other methods as a result on WebAssembly.
+	/// See https://github.com/dotnet/runtime/issues/56309
+	/// </remarks>
+	[MethodImpl(MethodImplOptions.AggressiveInlining)]
+	private void ReleasePointerNative()
+	{
+		try
+		{
+			_nativeCaptureElement.ReleasePointerNative(Pointer);
+		}
+		catch (Exception e)
+		{
+			this.Log().Error($"Failed to release native capture of {Pointer}", e);
```

Comment:
If the release fails, can we end up in an unrecoverable state?

Reply:
Yes and no : depends of the platform, but I would assume it would be only until the pointer is being released. Anyway this will impact only capture for which pointer goes outside of the bounds of the app.